### PR TITLE
IGAPP-407: Only deliver releases to production after testing in development

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

**Comment by klinzo** [comment]
I tested most of the lanes and commands already. Exception: Actual promotion from testflight to app store connect.

Promotion will fail for aschaffenburg if we don't create production releases beforehand but I think that will still take some time. For the ios release notes I had to make things a little bit hacky as one can't upload release notes and metadata for testflight builds.

File: .circleci/src/workflows/scheduled_delivery.yml [hunk 1]
```diff
@@ -249,7 +249,7 @@ jobs:
       context:
         - mattermost
         - deliverino
-      production_delivery: true
+      production_delivery: false
```
Comment (maxammann) [inline .circleci/src/workflows/scheduled_delivery.yml:252]: This flag is now always false, right? Should we remove that functionality?

**Review by maxammann — APPROVED** [review]

File: .circleci/src/workflows/scheduled_delivery.yml (hunk 1 above)
Comment (klinzo) [reply .circleci/src/workflows/scheduled_delivery.yml:252]: Not for triggered deliveries. I think we should still keep this as we may need it sometime.

**Review by Taggotty — APPROVED** [review]
Seems actually not that complicated. Although of course I am not that experienced with this release stuff.

**Pull request merged**